lnd.go

```diff
@@ -893,56 +893,52 @@ func Main(cfg *Config, lisCfg ListenerCfg, interceptor signal.Interceptor) error
 	// We transition the RPC state to Active, as the RPC server is up.
 	interceptorChain.SetRPCActive()
 
-	// If we're not in regtest or simnet mode, We'll wait until we're fully
-	// synced to continue the start up of the remainder of the daemon. This
-	// ensures that we don't accept any possibly invalid state transitions, or
-	// accept channels with spent funds.
-	if !(cfg.Bitcoin.RegTest || cfg.Bitcoin.SimNet ||
-		cfg.Litecoin.RegTest || cfg.Litecoin.SimNet) {
-
-		_, bestHeight, err := activeChainControl.ChainIO.GetBestBlock()
-		if err != nil {
-			err := fmt.Errorf("unable to determine chain tip: %v",
-				err)
-			ltndLog.Error(err)
-			return err
-		}
-
-		ltndLog.Infof("Waiting for chain backend to finish sync, "+
-			"start_height=%v", bestHeight)
-
-		for {
-			if !interceptor.Alive() {
-				return nil
-			}
-
-			synced, _, err := activeChainControl.Wallet.IsSynced()
-			if err != nil {
-				err := fmt.Errorf("unable to determine if "+
-					"wallet is synced: %v", err)
-				ltndLog.Error(err)
-				return err
-			}
+	// We'll wait until we're fully synced to continue the start up of the
+	// remainder of the daemon. This ensures that we don't accept any
+	// possibly invalid state transitions, or accept channels with spent
+	// funds.
+	_, bestHeight, err := activeChainControl.ChainIO.GetBestBlock()
```

Comment:
While I think it's correct to always wait to be synced, I'm still a bit worried about breaking too many people's test setups/assumptions with this. Alternatively we could just add an override flag here that we only set for our itests? Either as a config flag or a build tag specific variable.

Reply:
tbh I'm also concerned that this simple change might cause lots of unknown problems...however the build results look promising tho. I think if our setups/assumptions are based on an unsynced chain or wallet, I'm keen to fix those issues. Making a special case for `itest` imo decreases the efficacy of the tests. I'll be closely watching out for related effects from this change. WDYT?

Reply:
Yeah, you're right, adding another special case to the itests is probably not a good idea. I'm in favor of removing the condition and waiting for sync on all networks. @Roasbeef what's your opinion?

We will need to highlight this a bit more in the release notes, so an item somewhere in the `# Wallet` section that explains the change in a bit more detail.

Reply:
Updated!
Also do you think the `bitcoind`'s 2-hour-not-synced rule is configurable? Meaning can we change it to longer hours?

Reply:
Hmm, I don't think that's something that can be configured. It might be hard coded in and the two hours are actually the same time span that a block is allowed to be out of a node's time range to still be accepted?

I found this instead: https://github.com/lightningnetwork/lnd/issues/3336
@carlaKC do you remember why you opened that issue originally? Where did you encounter problems with a stale chain?

Reply:
I just ran into it occasionally in local dev when I was testing off-chain stuff and didn't mine a block in 2 hours. Seemed logical to add the exception for regtest since simnet was already there.

Reply:
In practice what I think will happen is that those running in regtest/simnet/signet, etc will need to _mine_a new block before `lnd` fully starts up. 

As is, it'll insta start up, but then users will need to mine a new block in order to have things flip to "synced to chain". 

Alternatively, couldn't we just apply this logic, but at the `lntest.NewNode` level? So that ends up blocking (using a similar loop as below) to ensure that all nodes are fully ready to go before being used within tests? So rather than making a special case within the itests, we instead just ensure that no node is actually usable (the initialization method will _block_) until the node is fully synced to the chain. 

Otherwise, I can see this change messing up things like Polar when users may have longer lived simnet chains with several nodes connected to them.

Reply:
A bit confused here...are you suggesting we keep the special case for regtest/simnet or remove it?

The problem with waiting in `lntest.NewNode` is that, in this case, it's not the itest that relies on it being synced, rather, it's the subserver that requires the wallet to be synced within `lnd`. The test failed because the sweeper started before the wallet was synced, thus the rescanning part was not finished and the sweeper was not aware of the UTXOs. This, however, wouldn't happen if it's on mainnet/testnet because we always require the wallet to finish syncing before moving forward, hence starting the server.

I think what's more important is that we want to have our itest best emulating the actual onchain conditions, yet we seem to have broken that promise by splitting another branch and build tests against it. They differ slightly and may have unintentional consequences, but I have the feeling the seeming nuance will bite us...

Reply:
A bit confused here...are you suggesting we keep the special case for regtest/simnet or remove it?

The problem with waiting in `lntest.NewNode` is that, in this case, it's not the itest that relies on it being synced, rather, it's the subserver that requires the wallet to be synced within `lnd`. The test failed because the sweeper started before the wallet was synced, thus the rescanning part was not finished and the sweeper was not aware of the UTXOs. This, however, wouldn't happen if it's on mainnet/testnet because we always require the wallet to finish syncing before moving forward, hence starting the server.

I think what's more important is that we want to have our itest best emulating the actual onchain conditions, yet we seem to have broken that promise by splitting another branch and build tests against it. They differ slightly and may have unintentional consequences, but I have the feeling the seeming nuance will bite us...